sentry/src/routes/channel.rs

```diff
@@ -846,4 +859,195 @@ mod test {
+            assert_eq!(
+                channels_list.channels.len(),
+                2,
+                "There should be 2 channels on the first page"
+            );
+            assert_eq!(
+                channels_list.channels, expected_channels_list,
+                "The channels should be listed by ascending order of their creation"
+            );
+            assert_eq!(
+                channels_list.pagination.total_pages, 2,
+                "There should be 2 pages in total"
+            );
+
+            let query = ChannelListQuery {
+                page: 1,
+                validator: None,
+            };
+            let res = channel_list(build_request(query), &app)
+                .await
+                .expect("should get channels");
+            let channels_list = res_to_channel_list_response(res).await;
+            let expected_channels_list = vec![channel_other_leader];
+            assert_eq!(
+                channels_list.channels.len(),
+                1,
+                "There should be 1 channel on the second page"
+            );
+            assert_eq!(
+                channels_list.channels, expected_channels_list,
+                "The channels should be listed by ascending order of their creation"
+            );
+        }
+
+        // Test query validator - query with validator ID
+        {
+            let query = ChannelListQuery {
+                page: 0,
+                validator: Some(IDS[&LEADER_2]),
+            };
+            let res = channel_list(build_request(query), &app)
+                .await
+                .expect("should get channels");
+            let channels_list = res_to_channel_list_response(res).await;
+            let expected_channels_list = vec![channel_other_leader];
+            assert_eq!(
+                channels_list.channels.len(),
+                1,
+                "There should be 1 channel in total"
+            );
+            assert_eq!(
+                channels_list.channels, expected_channels_list,
+                "Response returns the correct channel"
+            );
+            assert_eq!(
+                channels_list.pagination.total_pages, 1,
+                "There should be 1 page in total"
+            );
+        }
+        // Test query with both pagination and validator by querying for the follower validator
+        {
+            let query = ChannelListQuery {
+                page: 0,
+                validator: Some(IDS[&FOLLOWER]),
+            };
+            let res = channel_list(build_request(query), &app)
+                .await
+                .expect("should get channels");
+            let channels_list = res_to_channel_list_response(res).await;
+            let expected_channels_list = vec![channel, channel_other_token];
+            assert_eq!(
+                channels_list.channels.len(),
+                2,
+                "There should be 1 channel on the second page"
+            );
```

Comment:
These checks are redundant and will not provide enough information on failure.
The assertion with `Vec` will be enough and will provide additional information on missing or additional records that were not expected.

Also use `pretty_assertions`